src/WebUI/Components/Editor.tsx

```diff
@@ -0,0 +1,46 @@
+import { BaseComponent } from "@uifabric/utilities";
+import { IVssComponentProperties } from "../Types";
+import * as React from "react";
+// basic editor functionality
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+// additional functionality of editor
+import "monaco-editor/esm/vs/editor/browser/controller/coreCommands";
+import "monaco-editor/esm/vs/editor/contrib/find/findController";
+// languages supported in editor
+import "monaco-editor/esm/vs/basic-languages/yaml/yaml.contribution";
+
+
+export interface IEditorProps extends IVssComponentProperties {
+    options?: monaco.editor.IEditorConstructionOptions;
+}
+
+export class Editor extends BaseComponent<IEditorProps> {
+    constructor(props: IEditorProps) {
+        super(props, {});
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <div className="editor-outer" style={{ width: "100%", minHeight: "100px", margin: "10px 0px" }}>

```

Comment:
when we have the real usage, we can remove/change the stylings

Reply:
yes, extraEditorClassName is the option we can use for this. these styles can be removed once we know the exact use of this